Tryouts/Core/Services/ModulesService/ExecutableModule.cs

```diff
@@ -0,0 +1,98 @@
+﻿/// ********************************************************************************************************
+///
+/// Morgan Stanley makes this available to you under the Apache License, Version 2.0 (the "License").
+/// You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+/// See the NOTICE file distributed with this work for additional information regarding copyright ownership.
+/// Unless required by applicable law or agreed to in writing, software distributed under the License
+/// is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and limitations under the License.
+/// 
+/// ********************************************************************************************************
+
+using MorganStanley.ComposeUI.Tryouts.Core.Abstractions.Modules;
+using System.Diagnostics;
+
+namespace MorganStanley.ComposeUI.Tryouts.Core.Services.ModulesService;
+
+internal class ExecutableModule : ModuleBase
+{
+    private string _launchPath;
+    private Process? _mainProcess;
+    private bool exitRequested = false;
+
+    public ProcessInfo ProcessInfo => new ProcessInfo
```

Comment:
No, it is not to become modifiable

Reply:
I think it is better in any case to use a fixed object, rather that recreating it every time it is requested.

Reply:
Changed to struct.
Normally this is only queried when lifecycle events happen, and in that case it will have different data inside, but changing the contents could result in exposing inner behavior / unexpected side-effects, that's why a new instance is created every time.
I'm up for further discussion if you wish.

Reply:
I think I am good with it now. Since it is struct it will be copied by value and will not change on the client even if something changes on the server. That was my primary concern.